pytest_fixtures/certificate.py

```diff
@@ -0,0 +1,150 @@
+            'cert_file_name': cert_file_name,
+            'ca_bundle_file_name': ca_bundle_file_name,
+            'success_message': success_message,
+        }
+        return cert_data
+
+    return _cert_data
+
+
+@pytest.fixture()
+def cert_setup(cert_data):
+    """copy all configuration files to satellite host for generating custom certs"""
+
+    def _cert_setup(hostname=None):
+        # Need a subdirectory under ssl-build with same name as Capsule name
+        hostname = hostname or settings.server.hostname
+        _cert_data = cert_data(hostname)
+        with get_connection(timeout=100, hostname=hostname) as connection:
+            connection.run('mkdir ssl-build/{}'.format(_cert_data['capsule_hostname']))
+            # Ignore creation error, but assert directory exists
+            assert connection.run('test -e ssl-build/{}'.format(_cert_data['capsule_hostname']))
+        upload_file(
+            local_file=get_data_file('generate-ca.sh'),
+            remote_file='generate-ca.sh',
+            hostname=hostname,
+        )
+        upload_file(
+            local_file=get_data_file('generate-crt.sh'),
+            remote_file='generate-crt.sh',
+            hostname=hostname,
+        )
+        upload_file(
+            local_file=get_data_file('openssl.cnf'), remote_file='openssl.cnf', hostname=hostname
+        )
+        # create the CA cert.
+        with get_connection(timeout=300, hostname=hostname) as connection:
+            result = connection.run('echo 100001 > serial')
+            result = connection.run('bash generate-ca.sh')
+            assert result.return_code == 0
+        # create the Satellite's cert
+        with get_connection(timeout=300, hostname=hostname) as connection:
+            result = connection.run(
+                'yes | bash {} {}'.format('generate-crt.sh', _cert_data['sat6_hostname'])
+            )
+            assert result.return_code == 0
+
+    return _cert_setup
+
+
+@pytest.fixture()
+def certs_cleanup():
+    """cleanup all cert configuration files"""
+    yield
+    files = [
+        'cacert.crt',
+        'certindex*',
+        'generate-*.sh',
+        'capsule_cert' 'openssl.cnf',
+        'private',
+        'serial*',
+        'certs/*',
+        settings.server.hostname,
+    ]
+    with get_connection(timeout=300) as connection:
+        files = ' '.join(files)
+        result = connection.run(f'rm -rf {files}')
+        assert result.return_code == 0
+
+
+@pytest.fixture()
+def generate_certs():
+    """generate custom certs in satellite host"""
+    upload_file(
+        local_file=get_data_file('certs.sh'),
+        remote_file='certs.sh',
+    )
+    upload_file(
+        local_file=get_data_file('extensions.txt'),
+        remote_file='extensions.txt',
+    )
+    with get_connection(timeout=300) as connection:
+        result = connection.run('bash certs.sh')
+        assert result.return_code == 0
+
+
+@pytest.fixture(scope='module')
+def file_setup():
```

Comment:
These fixtures should have more descriptive names, since they're defined in a framework wide fixture module. It would be very easy for a fixture with this name to be defined in other modules in the `pytest_fixtures` directory.  Something like `capsule_cert_generate`